src/utils.ts

```diff
@@ -88,11 +88,32 @@ function escapeHTML(text: string): string {
   return text.replace(/[&"<>]/g, c => ({ '&': '&amp;', '"': '&quot;', '<': '&lt;', '>': '&gt;' }[c]!));
 }
 
-export async function spawnAsync(executable: string, args: string[], cwd?: string): Promise<string> {
+export function getPnPFilepath(workspaceDir: string): string {
+  return path.join(workspaceDir, '.pnp.cjs');
+}
+export function isPnPWorkspace(workspaceDir: string): boolean {
+  return fs.existsSync(getPnPFilepath(workspaceDir));
+}
+export function getPnPEnvVariables(workspaceDir: string): Record<string,string> {
+  if (!isPnPWorkspace(workspaceDir))
+    return {};
+
+  const pnpFile = getPnPFilepath(workspaceDir);
+  return {
+    NODE_OPTIONS: `--require ${pnpFile}`
```

Comment:
```suggestion
    NODE_OPTIONS: `${process.env.NODE_OPTIONS ? (process.env.NODE_OPTIONS + ' ') : ''}--require ${pnpFile}`
```

Reply:
Do you know if there could be other sources of `NODE_OPTIONS` other than `process.env`?
in `src/playwrightTests.ts` `debugTests`:
```js
  ...process.env,
  ...settingsEnv,
  ...this._reusedBrowser.browserServerEnv(),
  ...(await reporterServer.env()),
```
Can any of these provide additional `NODE_OPTIONS`? I think your suggestion could lead to options being applied twice

Reply:
NODE_OPTIONS can come from a variety of different places, `process.env` is very likely to have some, `settingsEnv` can also contain some. While the other two most likely not.

I think it's fine to merge all 3 into a single one, Node.js would deal with that in the end and prioritizes the right ones.

Reply:
I am just a little hesitant about merging all these as, in my understanding, only the last `NODE_OPTIONS` would survive?